src/app/domains/Form/routes/FormEdit/FormEdit.js

```diff
@@ -3,35 +3,73 @@ import {
   PageLayout,
   EditorSidebar,
   QuestionLayoutSwitcher,
-  FormContentArea
+  FormContentArea,
+  Spinner
 } from 'components'
 import { Box } from '@qonsoll/react-design'
 import { QuestionForm } from 'app/domains/Question/components'
-import { LAYOUT_TYPES, QUESTION_TYPES } from 'app/constants'
+import { QUESTION_TYPES, COLLECTIONS } from 'app/constants'
 import { LAYOUT_TYPE_KEYS } from 'app/constants/layoutTypes'
+import { useFormContext, useFormContextDispatch } from 'app/context/FormContext'
+import DISPATCH_EVENTS from 'app/context/FormContext/DispatchEventsTypes'
+import { useParams } from 'react-router'
+import {
+  useCollectionData,
+  useDocumentData
+} from 'react-firebase-hooks/firestore'
+import { getCollectionRef } from 'app/services/Firestore'
+
 // import PropTypes from 'prop-types'
-// import { useTranslation } from 'react-i18next'
 
 function FormEdit(props) {
   // const { WRITE_PROPS_HERE } = props
   // const { ADDITIONAL_DESTRUCTURING_HERE } = user
 
   // [ADDITIONAL HOOKS]
-  // const { t } = useTranslation('translation')
-  // const { currentLanguage } = t
+  const { id } = useParams()
+  const [form, formLoading] = useDocumentData(
+    getCollectionRef(COLLECTIONS.FORMS).doc(id)
+  )
+  const [questionsList, questionsListLoading] = useCollectionData(
+    getCollectionRef(COLLECTIONS.QUESTIONS).where('formId', '==', id)
+  )
 
-  // [COMPONENT STATE HOOKS]
-  const [activeKey, setActiveKey] = useState(LAYOUT_TYPE_KEYS[0])
-  const [questionType, setQuestionType] = useState(QUESTION_TYPES.YES_NO)
+  //[COMPONENT STATE HOOKS]
   const [showPopover, setshowPopover] = useState(false)
+
+  // [CUSTOM_HOOKS]
+  const formContext = useFormContext()
+  const dispatch = useFormContextDispatch()
+
   // [COMPUTED PROPERTIES]
+  let questions, endings
+  if (!formLoading && !questionsListLoading) {
+    questions = questionsList.filter(
+      (item) => item.questionType !== QUESTION_TYPES.ENDING
+    )
+
+    endings = questionsList.filter(
+      (item) => item.questionType === QUESTION_TYPES.ENDING
+    )
+  }
 
   // [CLEAN FUNCTIONS]
   const onChangeMenuItem = ({ key }) => {
-    setActiveKey(LAYOUT_TYPES[key])
+    dispatch({
+      type: DISPATCH_EVENTS.UPDATE_CURRENT_QUESTION,
+      payload: {
+        layoutType: key,
+        image:
+          formContext?.image ||
+          'https://www.awakenthegreatnesswithin.com/wp-content/uploads/2018/08/Nature-Quotes-1.jpg'
```

Comment:
Is it possible to move this URL to the const variavle?

Reply:
I`ve just created const, but forgot replace string to const